apps/api/src/services/interfaces/data-provider.interface.ts

```diff
@@ -7,6 +7,8 @@ import {
 } from './interfaces';
 
 export interface DataProviderInterface {
+  hasHistoricalData(symbol: string): boolean;
```

Comment:
I would rename it to something like `isActive(symbol: string): boolean;`, because it's not restricted to historical data, but if the data provider is used in general

Reply:
Maybe `isActiveForSymbol`? Since otherwise the parameter is confusing

Reply:
`canHandle` is fine! 👍🏻